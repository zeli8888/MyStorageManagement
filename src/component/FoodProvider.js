import React, { Component, useState, useEffect } from 'react';
import IngredientService from '../service/IngredientService'
import DishService from '../service/DishService'

export const FoodContext = React.createContext();  //exporting context object

const FoodProvider = ({ children }) => {
    const [ingredients, setIngredients] = useState([]);
    const [dishes, setDishes] = useState([]);

    useEffect(() => {
        IngredientService.getAllIngredients().then(response => { setIngredients(response.data) })
        DishService.getAllDishes().then(response => { setDishes(response.data) })
    }, []);

    return (
        <FoodContext.Provider value={
            {
                ingredients: ingredients,
                setIngredients: setIngredients,
                dishes: dishes,
                setDishes: setDishes
            }}>
            {children
                //this indicates that all the child tags with MyProvider as Parent can access the global store  
            }
        </FoodContext.Provider>)
}

export default FoodProvider;