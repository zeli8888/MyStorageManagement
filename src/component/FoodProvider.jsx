import React, { useState, useEffect } from 'react';
import ingredientService from '../service/IngredientService'
import dishService from '../service/DishService'

export const FoodContext = React.createContext();  //exporting context object

const FoodProvider = ({ children }) => {
    const [allIngredients, setAllIngredients] = useState([]);
    const [allDishes, setAllDishes] = useState([]);

    useEffect(() => {
        ingredientService.getAllIngredients().then(response => { setAllIngredients(response.data) }).catch(error => { console.log(error) });
        dishService.getAllDishes().then(response => { setAllDishes(response.data) }).catch(error => { console.log(error) });
    }, []);

    return (
        <FoodContext.Provider value={
            {
                allIngredients: allIngredients,
                setAllIngredients: setAllIngredients,
                allDishes: allDishes,
                setAllDishes: setAllDishes
            }}>
            {children
                //this indicates that all the child tags with MyProvider as Parent can access the global store  
            }
        </FoodContext.Provider>)
}

export default FoodProvider;