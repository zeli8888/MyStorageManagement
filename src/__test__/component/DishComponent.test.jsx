import { render, screen, waitFor, cleanup, fireEvent } from '@testing-library/react';
import { describe, test, vi, beforeEach, expect } from 'vitest';
import DishComponent from '../../component/DishComponent';
import { FoodContext } from '../../component/FoodProvider';
import dishService from '../../service/DishService';
import dishRecordService from '../../service/DishRecordService';
import '@testing-library/jest-dom/vitest';

// Mock navigation hook
vi.mock('react-router-dom', () => ({
    useNavigate: () => vi.fn(),
}));

// Mock service layer
vi.mock('../../service/DishService', () => ({
    default: {
        getAllDishes: vi.fn().mockResolvedValue({ data: [] }),
        searchDishes: vi.fn().mockResolvedValue({ data: [] }),
        addDish: vi.fn().mockResolvedValue({}),
        updateDish: vi.fn().mockResolvedValue({}),
        deleteDishes: vi.fn().mockResolvedValue({}),
    },
}));

vi.mock('../../service/DishRecordService', () => ({
    default: {
        addDishRecord: vi.fn().mockResolvedValue({}),
    },
}));

// Mock complex child components
vi.mock('../../component/utils', async (importOriginal) => ({
    ...await importOriginal(),
    EnhancedTableToolbar: ({ tableTitle, onSearch, deleteSelected, updateSelected }) => (
        <div>
            {tableTitle}
            <input data-testid="search-input" onChange={(e) => onSearch(e.target.value)} />
            <button data-testid="delete-button" onClick={deleteSelected}>Delete</button>
            <button data-testid="update-button" onClick={updateSelected}>Update</button>
        </div>
    ),
    DeletionConfirmationComponent: ({ open, onClose, onConfirm, warningMessage }) => (
        <div>
            {open && (
                <div>
                    <p>{warningMessage}</p>
                    <button onClick={onConfirm}>Confirm</button>
                    <button onClick={onClose}>Cancel</button>
                </div>
            )}
        </div>),
    EnhancedTableHead: () => <thead data-testid="enhanced-table-head" />,

}));

describe('Dish Component Integration Tests', () => {
    const mockDishes = [
        {
            dishId: 1,
            dishName: 'Test Dish',
            dishDesc: 'Test Description',
            dishIngredients: [
                {
                    dishIngredientQuantity: 2,
                    ingredient: {
                        ingredientId: 1,
                        ingredientName: 'Test Ingredient',
                        ingredientStorage: 5,
                        ingredientCost: 10,
                        ingredientDesc: 'Test Ingredient Description',
                    },
                },
            ],
        },
    ];

    const mockContextValue = {
        allIngredients: [
            {
                ingredientId: 1,
                ingredientName: 'Test Ingredient',
                ingredientStorage: 5,
                ingredientCost: 10,
                ingredientDesc: 'Test Ingredient Description',
            },
        ],
        setAllDishes: vi.fn(),
    };

    beforeEach(() => {
        vi.clearAllMocks();
        dishService.getAllDishes.mockResolvedValue({ data: mockDishes });
        dishService.searchDishes.mockResolvedValue({ data: mockDishes });
        dishService.addDish.mockResolvedValue({});
        dishService.updateDish.mockResolvedValue({});
        dishService.deleteDishes.mockResolvedValue({});
    });

    afterEach(cleanup);

    // Base snapshot verification
    test('initial render matches snapshot', async () => {
        const { asFragment } = render(
            <FoodContext.Provider value={mockContextValue}>
                <DishComponent />
            </FoodContext.Provider>
        );

        await waitFor(() => expect(screen.getByText('Test Dish')).toBeInTheDocument());
        expect(asFragment()).toMatchSnapshot();
    });

    // Search functionality
    test('search triggers API call', async () => {
        render(
            <FoodContext.Provider value={mockContextValue}>
                <DishComponent />
            </FoodContext.Provider>
        );

        await waitFor(() => screen.getByTestId('search-input'));
        fireEvent.change(screen.getByTestId('search-input'), { target: { value: 'test' } });

        await waitFor(() => {
            expect(dishService.searchDishes).toHaveBeenCalledWith('test');
        });
    });

    // Add functionality
    test('adds new dish with ingredients', async () => {
        render(
            <FoodContext.Provider value={mockContextValue}>
                <DishComponent />
            </FoodContext.Provider>
        );

        fireEvent.click(screen.getByText(/New Dish/i));
        fireEvent.change(screen.getByLabelText(/Dish Name/i), { target: { value: 'New Dish' } });
        const select = screen.getByRole('combobox', { name: /Recipe/i });
        fireEvent.click(select);
        fireEvent.click(screen.getByText(/Save/i));

        await waitFor(() => {
            expect(dishService.addDish).toHaveBeenCalled();
        });
    });

    // Update functionality
    test('existing dish update', async () => {
        render(
            <FoodContext.Provider value={mockContextValue}>
                <DishComponent />
            </FoodContext.Provider>
        );
        const checkboxes = await waitFor(() => screen.getAllByRole('checkbox'));
        fireEvent.click(checkboxes[1]); // Select first dish
        fireEvent.click(screen.getByTestId(/update-button/i));
        fireEvent.change(screen.getByLabelText(/Dish Name/i), { target: { value: 'Updated' } });
        fireEvent.click(screen.getByText(/Save/i));

        await waitFor(() => {
            expect(dishService.updateDish).toHaveBeenCalledWith(
                1,
                expect.objectContaining({
                    dish: expect.objectContaining({
                        dishName: "Updated"
                    })
                })
            )
        });
    });


    // Delete functionality
    test('deletes selected dishes', async () => {
        render(
            <FoodContext.Provider value={mockContextValue}>
                <DishComponent />
            </FoodContext.Provider>
        );

        await waitFor(() => screen.getByText('Test Dish'));
        const checkboxes = screen.getAllByRole('checkbox');
        fireEvent.click(checkboxes[1]); // Select first dish
        fireEvent.click(screen.getByTestId('delete-button'));
        fireEvent.click(screen.getByText(/confirm/i));

        await waitFor(() => {
            expect(dishService.deleteDishes).toHaveBeenCalled();
        });
    });

    // Record addition
    test('adds dish record', async () => {
        render(
            <FoodContext.Provider value={mockContextValue}>
                <DishComponent />
            </FoodContext.Provider>
        );

        await waitFor(() => screen.getByText('Test Dish'));
        fireEvent.click(screen.getByRole('button', { name: /add/i }));
        fireEvent.click(screen.getByText('Save'));

        await waitFor(() => {
            expect(dishRecordService.addDishRecord).toHaveBeenCalled();
        });
    });

    // Error state snapshot
    test('error alert', async () => {
        const consoleErrorMock = vi.spyOn(console, 'log').mockImplementation(() => { });
        dishService.getAllDishes.mockRejectedValue(new Error('test Error'));
        let mockSetAllDishes = vi.fn();

        render(
            <FoodContext.Provider value={{ setAllDishes: mockSetAllDishes, allIngredients: [] }}>
                <DishComponent />
            </FoodContext.Provider>
        );

        await waitFor(() => {
            expect(mockSetAllDishes).not.toHaveBeenCalled();
        });
        expect(consoleErrorMock).toHaveBeenCalledWith(new Error('test Error'));
        consoleErrorMock.mockRestore();
    });

    // Expandable content
    test('displays recipe details when expanded', async () => {
        render(
            <FoodContext.Provider value={mockContextValue}>
                <DishComponent />
            </FoodContext.Provider>
        );

        await waitFor(() => screen.getByText('Test Dish'));
        fireEvent.click(screen.getByRole('button', { name: /expand row/i }));

        await waitFor(() => {
            expect(screen.getByText('Test Ingredient')).toBeInTheDocument();
        });
    });
});