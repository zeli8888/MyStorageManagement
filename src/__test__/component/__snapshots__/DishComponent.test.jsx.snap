// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dish Component Integration Tests > displays recipe details when expanded 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-i9gxme"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-d64g0g-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
      />
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
      >
        <div>
          Dishes
          <input
            data-testid="search-input"
          />
          <button
            data-testid="delete-button"
          >
            Delete
          </button>
          <button
            data-testid="update-button"
          >
            Update
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-y2ff7i-MuiPaper-root-MuiTableContainer-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <table
            aria-label="collapsible table"
            class="MuiTable-root css-4z2v8j-MuiTable-root"
          >
            <thead
              data-testid="enhanced-table-head"
            />
            <tbody
              class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
            >
              <tr
                aria-checked="false"
                class="MuiTableRow-root MuiTableRow-hover css-6ge3s-MuiTableRow-root"
                role="checkbox"
                tabindex="-1"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-mfzv8e-MuiTableCell-root"
                >
                  <span
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-vuhw6e-MuiButtonBase-root-MuiSwitchBase-root-MuiCheckbox-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
                      data-indeterminate="false"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                </td>
                <th
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                  scope="row"
                >
                  Test Dish
                </th>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  Test Description
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInfo MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInfo css-mnflth-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Add
                  </button>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <button
                    aria-label="expand row"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-155wbr6-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="KeyboardArrowUpIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z"
                      />
                    </svg>
                  </button>
                </td>
              </tr>
              <tr
                class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                  colspan="5"
                  style="padding-bottom: 0px; padding-top: 0px;"
                >
                  <div
                    class="MuiCollapse-root MuiCollapse-vertical css-1f6q371-MuiCollapse-root"
                    style="min-height: 0px; height: 0px; transition-duration: 0ms; transition-timing-function: undefined;"
                  >
                    <div
                      class="MuiCollapse-wrapper MuiCollapse-vertical css-1x6hinx-MuiCollapse-wrapper"
                    >
                      <div
                        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-1i4ywhz-MuiCollapse-wrapperInner"
                      >
                        <div
                          class="MuiBox-root css-1ynyhby"
                        >
                          <div
                            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-gnhzar-MuiTypography-root"
                          >
                            Recipe
                          </div>
                          <table
                            aria-label="purchases"
                            class="MuiTable-root css-1xwxv7r-MuiTable-root"
                          >
                            <thead
                              class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
                            >
                              <tr
                                class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
                              >
                                <th
                                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-g6bqb5-MuiTableCell-root"
                                  scope="col"
                                >
                                  Name
                                </th>
                                <th
                                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-g6bqb5-MuiTableCell-root"
                                  scope="col"
                                >
                                  Amount
                                </th>
                                <th
                                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-g6bqb5-MuiTableCell-root"
                                  scope="col"
                                >
                                  Storage
                                </th>
                                <th
                                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-g6bqb5-MuiTableCell-root"
                                  scope="col"
                                >
                                  Cost
                                </th>
                                <th
                                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-g6bqb5-MuiTableCell-root"
                                  scope="col"
                                >
                                  Description
                                </th>
                              </tr>
                            </thead>
                            <tbody
                              class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
                            >
                              <tr
                                class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
                              >
                                <th
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-jiabcu-MuiTableCell-root"
                                  scope="row"
                                >
                                  Test Ingredient
                                </th>
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-jiabcu-MuiTableCell-root"
                                >
                                  2
                                </td>
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-jiabcu-MuiTableCell-root"
                                >
                                  5
                                </td>
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-jiabcu-MuiTableCell-root"
                                >
                                  10
                                </td>
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-jiabcu-MuiTableCell-root"
                                >
                                  Test Ingredient Description
                                </td>
                              </tr>
                            </tbody>
                          </table>
                        </div>
                      </div>
                    </div>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="MuiTablePagination-root css-7mt0f-MuiTablePagination-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-1gak8h1-MuiToolbar-root-MuiTablePagination-toolbar"
          >
            <div
              class="MuiTablePagination-spacer css-1wtxofq-MuiTablePagination-spacer"
            />
            <p
              class="MuiTablePagination-selectLabel css-s09cke-MuiTablePagination-selectLabel"
              id=":r33:"
            >
              Rows per page:
            </p>
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-select MuiSelect-root MuiTablePagination-input css-oegngy-MuiInputBase-root-MuiTablePagination-select"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby=":r33: :r32:"
                class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-18zjd9i-MuiNativeSelect-root-MuiSelect-select-MuiInputBase-input"
                id=":r32:"
                role="combobox"
                tabindex="0"
              >
                10
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
                tabindex="-1"
                value="10"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-ug7h84-MuiSvgIcon-root-MuiNativeSelect-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              class="MuiTablePagination-displayedRows css-11cfq65-MuiTablePagination-displayedRows"
            >
              1–1 of 1
            </p>
            <div
              class="MuiTablePaginationActions-root MuiTablePagination-actions css-sjsoon-MuiTablePaginationActions-root"
            >
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-h7cjts-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to previous page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="KeyboardArrowLeftIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </button>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-h7cjts-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to next page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="KeyboardArrowRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess css-qlt66a-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        New Dish
      </button>
    </div>
    <div />
  </div>
</DocumentFragment>
`;

exports[`Dish Component Integration Tests > initial render matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-i9gxme"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-d64g0g-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
      />
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
      >
        <div>
          Dishes
          <input
            data-testid="search-input"
          />
          <button
            data-testid="delete-button"
          >
            Delete
          </button>
          <button
            data-testid="update-button"
          >
            Update
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-y2ff7i-MuiPaper-root-MuiTableContainer-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <table
            aria-label="collapsible table"
            class="MuiTable-root css-4z2v8j-MuiTable-root"
          >
            <thead
              data-testid="enhanced-table-head"
            />
            <tbody
              class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
            >
              <tr
                aria-checked="false"
                class="MuiTableRow-root MuiTableRow-hover css-6ge3s-MuiTableRow-root"
                role="checkbox"
                tabindex="-1"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-mfzv8e-MuiTableCell-root"
                >
                  <span
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-vuhw6e-MuiButtonBase-root-MuiSwitchBase-root-MuiCheckbox-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
                      data-indeterminate="false"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                </td>
                <th
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                  scope="row"
                >
                  Test Dish
                </th>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  Test Description
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInfo MuiButton-root MuiButton-contained MuiButton-containedInfo MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInfo css-mnflth-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Add
                  </button>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <button
                    aria-label="expand row"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-155wbr6-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="KeyboardArrowDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
                      />
                    </svg>
                  </button>
                </td>
              </tr>
              <tr
                class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                  colspan="5"
                  style="padding-bottom: 0px; padding-top: 0px;"
                />
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="MuiTablePagination-root css-7mt0f-MuiTablePagination-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-1gak8h1-MuiToolbar-root-MuiTablePagination-toolbar"
          >
            <div
              class="MuiTablePagination-spacer css-1wtxofq-MuiTablePagination-spacer"
            />
            <p
              class="MuiTablePagination-selectLabel css-s09cke-MuiTablePagination-selectLabel"
              id=":r1:"
            >
              Rows per page:
            </p>
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-select MuiSelect-root MuiTablePagination-input css-oegngy-MuiInputBase-root-MuiTablePagination-select"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby=":r1: :r0:"
                class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-18zjd9i-MuiNativeSelect-root-MuiSelect-select-MuiInputBase-input"
                id=":r0:"
                role="combobox"
                tabindex="0"
              >
                10
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
                tabindex="-1"
                value="10"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-ug7h84-MuiSvgIcon-root-MuiNativeSelect-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              class="MuiTablePagination-displayedRows css-11cfq65-MuiTablePagination-displayedRows"
            >
              1–1 of 1
            </p>
            <div
              class="MuiTablePaginationActions-root MuiTablePagination-actions css-sjsoon-MuiTablePaginationActions-root"
            >
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-h7cjts-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to previous page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="KeyboardArrowLeftIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </button>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-h7cjts-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to next page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="KeyboardArrowRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess css-qlt66a-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        New Dish
      </button>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
